### PR TITLE
Fix handling of Chunked requests

### LINKED ISSUE #1720: superset login is not working when Transfer encoding set to chunked in http request
I am trying to use superset behind a proxy server, where the proxy server is using chunked encoding in requests. After debugging for a while found that when using chunked encoding the login request  ignores the request content. 

Make sure these boxes are checked before submitting your issue - thank you!

- [*] I have checked the superset logs for python stacktraces and included it here as text if any
- [*] I have reproduced the issue with at least the latest released version of superset
- [*] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.14.1

### Steps to reproduce
disable CSRF by setting WTF_CSRF_ENABLED to false. 

1) try to login via curl below command works and login works fine- 
curl -X POST 'http://<superset-ip:port>/login/'  --data 'csrf_token=None&username=admin&password=admin’

2) set  encoding to chunked , now login will not work  - 
curl -X POST 'http://<superset-ip:port>/login/' -H 'Transfer-Encoding: chunked' --data 'csrf_token=None&username=admin&password=admin'

### PR DESCRIPTION
**Pull request by nishantmonu51**
Add fix for handling chunk encoding requests.
If ENABLE_CHUNK_ENCODING is set to true, for requests with transfer
encoding set to true. It will set wsgi.input_terminated to true which
tells werkzeug to ignore content-length and read the stream till the
end.

Fixes https://github.com/airbnb/superset/issues/1720

### CODE REVIEW

File: superset/__init__.py [hunk 1]
```diff
@@ -58,6 +58,21 @@
...
+if app.config.get('ENABLE_CHUNK_ENCODING'):
+    class ChunkedEncodingFix(object):
+
+        def __init__(self, app):
+            self.app = app
+
+        def __call__(self, environ, start_response):
+            # Setting wsgi.input_terminated tells werkzeug.wsgi to ignore
+            # content-length and read the stream till the end.
+            print(environ)
```
Comment (xrmx) [inline superset/__init__.py]: leftover?

Comment (nishantmonu51) [inline superset/__init__.py]: fixed.

**Comment by nishantmonu51** [comment]
@xrmx Any more comments ?

**Comment by nishantmonu51** [comment]
@xrmx: Any more comments on this ?

**Comment by xrmx** [comment]
@nishantmonu51 nope, looks fine to me

**Comment by nishantmonu51** [comment]
@xrmx: can this be merged then ? do i need to contact someone else for getting this in ?

**Comment by nishantmonu51** [comment]
@mistercrunch: please review.

**Comment by mistercrunch** [comment]
LGTM
@nishantmonu51 I granted you write rights on the repo, please confirm by merging the PR!

**Pull request merged**